src/widget/wtrackmenu.h

```diff
@@ -43,15 +44,17 @@ class WTrackMenu : public QMenu {
         HideUnhidePurge = 1 << 9,
         FileBrowser = 1 << 10,
         Properties = 1 << 11,
+        SearchRelated = 1 << 12,
         TrackModelFeatures = Remove | HideUnhidePurge,
         All = AutoDJ | LoadTo | Playlist | Crate | Remove | Metadata | Reset |
-                BPM | Color | HideUnhidePurge | FileBrowser | Properties
+                BPM | Color | HideUnhidePurge | FileBrowser | Properties |
+                SearchRelated
     };
     Q_DECLARE_FLAGS(Features, Feature)
 
     WTrackMenu(QWidget* parent,
             UserSettingsPointer pConfig,
-            TrackCollectionManager* pTrackCollectionManager,
+            Library* pLibrary,
```

Comment:
Increasing coupling between WTrackMenu and Library smells like not a great design decision to me. Could WTrackMenu instead emit a signal to trigger the query?

Reply:
Hmm, I'm not sure that would really make a difference though. The parent of WTrackMenu would still need a Library* pointer to make the signal-slot connection. What do you think?

Reply:
I agree, but since we already need either `Library` or `TrackCollectionManager` passing the top-level `Library` is even easier and cleaner now.

The new WSearchRelatedTracksMenu class will emit a signal that is received by WTrackMenu. The function invocation on `Library` will remain. I am not refactoring the whole design in this PR.

Reply:
okay